packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,191 +1,71 @@
-import React, { useEffect, useRef, useState } from "react";
-import classnames from "classnames";
-import styles from "./Combobox.css";
-import { Icon } from "../Icon";
-import { Text } from "../Text";
-import { Typography } from "../Typography";
-import { Button } from "../Button";
-
-interface ComboboxProps {
-  readonly children: React.ReactNode;
-}
-
-interface ContentProps {
-  /**
-   * options for the combobox
-   */
-  readonly options: string[];
-
-  /**
-   * onSelection handler.
-   */
-  onSelection: (selection: string) => void;
-
-  /**
-   * actions
-   */
+import React, { ReactNode } from "react";
+import { useAssert } from "@jobber/hooks/useAssert";
+import { Content } from "./components/Content";
+import { Action } from "./components/Action";
+import { TriggerButton, TriggerChip } from "./components/Trigger";
+import { ComboboxContextProvider } from "./ComboboxProvider";
+
+export interface ComboboxProps {
   readonly children: React.ReactNode;
```

Comment:
you're not expecting a random string to be put in as children, yeah? If so, 

```suggestion
  readonly children: ReactElement | ReactElement[];
```

so you won't have to check if the `child` is not a string

Reply:
true we're not expecting it no, but there's nothing stopping someone from doing so

I feel... I dunno, dishonest? about telling Typescript "hey this will only be this thing" when I can't possibly guarantee what someone might pass as `children`

if I promise TS that it'll only ever be an element, and we code around that but then we do get a string. it will most certainly break on something. 

so to me the question becomes how much do we want to protect people from themselves? if someone gives us nonsense and the component falls flat on its face, is that fine? or should we take measures to more elegantly handle incorrect input?

Reply:
Typescript will stop someone from doing that no? Like, that is the point of typescript. We said "only elements" and consumer said "nah, I want a string." The component should fail at that point if you intentionally suppress TS since we specifically said element only please.

IMO, supporting unsupported behaviour would create more problems and bugs for us.

Reply:
k I replied to you but changed my mind because I didn't understand something correctly. I think your suggestion makes sense! I'd even be tempted to go one step further with it and say

`children: ReactElement<ContentProps> | ReactElement<ContentProps>[] | ReactElement<TriggerProps> | ReactElement<TriggerProps>[]`

idk, that might be overkill. I'll give it a quick try here.

Reply:
That works too! Only advantage there is if you do drill down on it, you get the correct type of props

Reply:
Could also do the typescript way of recursive typing. Cmd + click ReactNode and see how it declared both single and array type without ever typing `this | this[]` 😉